src/NHibernate/Proxy/FieldInterceptorProxyBuilder.cs

```diff
@@ -16,8 +16,10 @@ internal static class FieldInterceptorProxyBuilder
 		private static readonly PropertyInfo AccessorTypeFieldInterceptorProperty =
 			FieldInterceptorAccessorType.GetProperty(nameof(IFieldInterceptorAccessor.FieldInterceptor));
 		private static readonly System.Type FieldInterceptorType = typeof(IFieldInterceptor);
+		private static readonly System.Type FieldInterceptorExtensionsType = typeof(FieldInterceptorExtensions);
 		private static readonly MethodInfo FieldInterceptorInterceptMethod = FieldInterceptorType.GetMethod(nameof(IFieldInterceptor.Intercept));
```

Comment:
Well, I have mixed feelings on this subject, because the obsolete code is this IL generation case is nearer to what we should have in 6.0 than the non-obsoleted code.
But it feels as a bug than it compiles without failing due to referencing an obsolete symbol. Rider shows it as an error, but the compiler is fine with it. That is maybe a compiler bug for the `nameof` case when it refers an obsolete property. In such case the compilation may break on a tooling update.

Reply:
I did a [quick test](https://dotnetfiddle.net/WVCpzK) and it seems that for calling a no-op `Intercept` method is twice faster without the cast for a million calls, which is not a big deal as casting a million times takes approx. 9ms. So we could remove it in order to avoid showning as an error in Rider. It seems that the compiler is not smart enough to detect that the only reference that the `nameof` references is an obsolete member.